packages/engine-backend/router/resourceRouter.ts

```diff
@@ -189,21 +235,77 @@ export const resourceRouter = trpc.router({
           endUserId: zEndUserId.nullish(),
           connectorConfigId: zId('ccfg').nullish(),
           connectorName: z.string().nullish(),
+          refreshToken: z.boolean().optional(),
+          forceRefresh: z.boolean().optional(),
         })
         .optional(),
     )
     .output(z.array(zRaw.resource))
     .query(async ({input = {}, ctx}) => {
-      const resources =
+      let resources =
         await ctx.services.metaService.tables.resource.list(input)
+
+      // Handle forceRefresh for each resource
+      
+      console.log('[listResources] Refreshing tokens for all resources');
+      const updatedResources = await Promise.all(resources.map(async (reso) => {
+        // @ts-expect-error
+        const expiresAt = reso?.settings?.['oauth']?.credentials?.raw?.expires_at
+
+        if (expiresAt && (input.forceRefresh || new Date(expiresAt).getTime() <= Date.now())) {
+          console.log(`[listResources] Refreshing token for resource ${reso.connectorName}`);
+          const resoCheck = await performResourceCheck(ctx, reso.id, {});
+          if(!resoCheck) {
+            console.warn(`[listResources] resourceCheck not implemented for ${reso.connectorName} which requires a refresh. Returning the stale resource.`);
+          }
+          return resoCheck || reso;
+        }
+        return reso;
+      }));
+
+      resources = updatedResources;
+      
+      // Remove refresh_token from the response for security reasons if not explicitly requested
```

Comment:
Feels like a hack. Doesn't feel worth it as is.

Reply:
i agree. should i remove?

Reply:
yea I'd just get rid of it for now to reduce complexity and we can add a follow up to properly implement it later

Reply:
Removed.